### PR TITLE
Editor view options

### PR DESCRIPTION
**Pull request by 7Jaws9Eyes**
Added a bunch of configurable viewing options to the side bar, including keyboard shortcuts for toggles.
Could be useful to make selecting splats more precise, especially if more editing tools are added in the future.

Sidebar with new options:
image

Center point colors and alpha value can be changed. Same for selected center points.
Keyboard shortcut J toggles center point alpha.
Keyboard shortcut L toggles alpha of selected center points.
(image below shows 2 custom colors and alpha values for center points)
image

Color of splats in selection can be interpolated with custom color. (interpolation 0 leaves color of selected splats unchanged).
K toggles selected splat coloration. 
image

Bounding rings of splats can be enabled in the centers view mode -> centers and bounds in same view. Can also be only enabled for selected splats.
Keyboard shortcut O toggles bounding rings. Q toggles only selected splats bounding rings.
image

Splat rendering can be disabled, shows only center points so scene is viewable as a point cloud. N toggles splat rendering 
image

Rings view mode uses bounding rings slider for rings size but maybe that mode could be removed as setting center points alpha to zero and enabling bounding rings gives the same view essentially.

### CODE REVIEW

File: src/ui/control-panel.ts [hunk 1]
```diff
@@ -66,6 +67,216 @@ class ControlPanel extends Panel {
...
+            text: 'Center Point Alpha'
+        });
+
+        const centerPointAlphaSlider = new SliderInput({
+            class: 'control-element-expand',
+            precision: 2,
+            min: 0,
+            max: 0.5,
+            value: 0.5
+        });
```
Comment (willeastcott) [inline src/ui/control-panel.ts:141]: Why have a separate slider for alpha here? You can pass `components: 4` to the `ColorPicker` constructor to add support for an alpha component.

Comment (7Jaws9Eyes) [reply src/ui/control-panel.ts:141]: i see, didnt know. i will implement it like that :)

Comment (7Jaws9Eyes) [reply src/ui/control-panel.ts]: i replaced both alpha sliders and the interpolation slider with alpha channels in color pickers

**Comment by willeastcott** [comment]
I think these settings belong in the OPTIONS section of the UI panel. Can you move them there and update the screenshot in the PR description please?

**Comment by 7Jaws9Eyes** [comment]
done.
do you think keeping "mode" option in camera section is still useful? as you can just enable the rings in the options now anyway.
and should "splat size" be moved to options section too?

**Comment by willeastcott** [comment]
I think Mode belongs in the SELECTION section actually. The UI does need a bit of a redesign, I think. I'm meeting someone about it today. 😄 

Sorry about the changes we've made since you submitted your PR - there are a couple of files with conflicts to resolve.

**Comment by 7Jaws9Eyes** [comment]
no problem, ill adapt it further when you request it. 
should i merge my branch with your main branch on my fork and push it to this PR or would you prefer to do it?

**Comment by willeastcott** [comment]
If you could merge from the main branch into your fork, that'd be great!

**Comment by 7Jaws9Eyes** [comment]
done :)

File: src/splat-debug.ts [hunk 2]
```diff
@@ -54,6 +49,7 @@ class SplatDebug {
     splatData: GSplatData;
     meshInstance: MeshInstance;
     size = 2;
+    // color = [0.0, 0.0, 1.0]
```
Comment (willeastcott) [inline src/splat-debug.ts:52]: Delete

File: src/splat.ts [hunk 3]
```diff
@@ -55,9 +59,18 @@ flat varying highp uint vertexState;
...
+uniform vec4 selectedSplatColor;
+uniform bool selectedSplatRingsToggle;
+uniform float selectedSplatRingsSize;
+uniform bool boundingRingToggle;
+uniform float boundingRingSize;
+uniform bool splatDisplayToggle;
 
 void main(void)
 {
+    if(splatDisplayToggle){
```
Comment (willeastcott) [inline src/splat.ts:71]:
```suggestion
    if (splatDisplayToggle){
```

File: src/ui/control-panel.ts [hunk 4]
```diff
@@ -647,7 +911,7 @@ class ControlPanel extends Panel {
 
         events.on('splat.count', (count: number) => {
             selectionPanel.headerText = `SELECTION${count === 0 ? '' : ' (' + count.toString() + ')'}`;
-        });
+        });
```
Comment (willeastcott) [inline src/ui/control-panel.ts:914]:
```suggestion
        });
```

File: src/main.ts [hunk 5]
```diff
@@ -78,24 +78,83 @@ const initShortcuts = (events: Events) => {
     shortcuts.register(['U', 'u'], { event: 'select.unhide' });
     shortcuts.register(['['], { event: 'tool.brushSelection.smaller' });
     shortcuts.register([']'], { event: 'tool.brushSelection.bigger' });
+    shortcuts.register(['9'], { event: 'tool.brushSelection.smaller' });
+    shortcuts.register(['0'], { event: 'tool.brushSelection.bigger' });
     shortcuts.register(['Z', 'z'], { event: 'edit.undo', ctrl: true });
     shortcuts.register(['Z', 'z'], { event: 'edit.redo', ctrl: true, shift: true });
     shortcuts.register(['M', 'm'], { event: 'camera.toggleMode' });
-
+
```
Comment (willeastcott) [inline src/main.ts:86]:
```suggestion

```

Comment (willeastcott) [inline src/main.ts:86]:
```suggestion

```

File: src/main.ts [hunk 6]
```diff
@@ -78,24 +78,83 @@ const initShortcuts = (events: Events) => {
...
+    
     // keep tabs on splat size changes
     let splatSizeSave = 2;
     events.on('splatSize', (size: number) => {
         if (size !== 0) {
             splatSizeSave = size;
         }
     });
-
+
```
Comment (willeastcott) [inline src/main.ts:94]:
```suggestion

```

**Review by willeastcott — APPROVED** [review]
Thanks for making all the requested changes, @7Jaws9Eyes. So consider my review a first pass - we still need feedback and a review from @slimbuck. He's out this week but will be back on Monday.

**Comment by slimbuck** [comment]
This was eventually done in #258

**Pull request closed without merging**